### DISCUSSION TITLE
module './node_modules/lit/index.js' does not provide an export named 'ReactiveController'

### DISCUSSION THREAD

**Discussion by anneb** [lit/lit#3872] [Q&A] [answered]
According to the documentation at https://lit.dev/docs/api/controllers/#ReactiveController, the following should be possible:

import { ReactiveController } from 'lit';

However, in my vanilla Javascript project this results in a Browser console error message:
```
module './node_modules/lit/index.js' does not provide an export named 'ReactiveController'
````

I have created a minimal git repository to demo the problem at https://github.com/anneb/lit-import-problem
I am using 

```bash
node --version
v18.6.0
````

What am I doing wrong?

**Comment by tbroyer** [comment, marked as answer]
`ReactiveController` is a _type_, so it only makes sense for TypeScript. In vanilla JS you'd use "duck-typing", and if/when you use JSDoc to document your code you could [import it](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#import-types) as `@typedef { import("./node_modules/lit/index").ReactiveController } ReactiveController`

**Reply by justinfagnani** [reply]
Thanks for the answer @tbroyer!

Yeah, this could maybe be more obvious in the docs, but `ReactiveController` is labeled with **type** here: 
![Screenshot 2023-05-04 at 9 03 48 AM](https://user-images.githubusercontent.com/522948/236262072-34330ce9-939a-43c5-b274-6f5d9c202739.png)

We're using TypeDoc to generate the API docs. Maybe there's a way to can label type imports as TypeScript-only?
